aip/client-libraries/4221.md

```diff
@@ -0,0 +1,80 @@
+---
+aip:
+  id: 4221
+  scope: client-libraries
+  state: draft
+  created: 2019-06-17
+permalink: /client-libraries/4221
+redirect_from:
+  - /4221
+---
+
+# Client-side retry
+
+Many APIs have error modes that a client should retry. These error modes vary
+across APIs, leaving users to hand-write retry logic around client libraries
+based on available documentation.
+
+## Guidance
+
+Client libraries **may** provide automatic client-side retry logic. Client
+libraries with automatic client-side retry logic **should** provide mechanisms
+(e.g. client initialization options) for users to configure it.
+
+Client libraries for API systems that support remotely-resolved client retry 
+configuration **should** respect the remotely-resolved configuration. However,
+user configuration via the aforementioned mechanisms **must** be respected.
+
+Client library generators implementing this feature **must** accept a retry
+configuration. This retry configuration **may** be supplied via a `protoc`
+plugin option. In the absence of a given retry configuration, client library
+generators **should not** generate a default retry configuration.
+
+### Retry implementation
+
+Client libraries **should** make client-side retry transparent to the user. The
+user **should not** have to opt-in to client-side retry explicitly, but the user
+**must** have a way to disable client-side retry altogether.
+
+### Retry configuration mechanisms
+
+Client libraries **should** surface a mechanism through which users may control
+the client-side retry configuration, including being able to disable client-side
```

Comment:
s/including being able to disable/including disabling

Reply:
Done.